### ISSUE TITLE
Add support to determine a token's index in the registry

### ISSUE THREAD

**Issue by pstehlik** [centrifuge/centrifuge-ethereum-contracts#137]
The openzeppelin implementation of ERC721 does not expose a way to find the token index for a specific token ID.

As a user of protocols like Dharma I want to be able to rely on the token's index when interacting with the Centrifuge Token Contracts. 
I want to do that because dharma (and presumably others) integer sizes smaller than uint256 in their implementations.
In the Dharma case, the debt order has less space for the params than uint256.

So I want to use the methods that the Enumberable Extension for ERC721 supports (e.g. `tokenByIndex(uint256 index)`).
To do so, I need to be able to find out the current index of a specific token before I construct the (in this example) Dharma debt order.

### currentIndexOfToken
* Rely on the internal `_allTokensIndex` mapping which tracks tokenId->index in the openzeppelin implementation
* add a new method `currentIndexOfToken(uint256 tokenId) public view returns (uint256)` to the UserMintableERC721 base contract that returns the current index of a token
* Be aware: `_allTokensIndex` **CHANGES** when tokens get burned (positions of tokens get swapped). This is critical to document as the index of a specific token might change. The caller of this new `currentIndexOfToken` will have to call this method each time they want to find out the token's current index. In the case of the current implementations of Centrifuge NFTs there is no `burn` method exposed so this edge-case will not be hit, but something to document clearly.

## Emit event upon minting that contains the token's index
* emit an event upon minting that includes the token's index. 
  * Add the index to the already triggered event when minting
* Calling systems (i.e. a Centrifuge Node) can then take the event from the `mint` call and rely on the token index that was given at that point and return it

**Comment by Remi-Tuy** [comment]
We have these requirements specifically, let either @Przemek-Win or my self know if we can help in anyways.

**Comment by rdinicut** [comment]
All mappings in ERC721Enumerable are private so in order to implement this there are 2 options
1. Store the index in InvoiceUnpaidNFT in tokenDetails
2. ERC721Enumerable and add currentIndexOfToken to it. 

This was implemented with forking because we did not want to increase gasPrice for mint by adding another uint256 value to the TokenDetails Struct and also we want that all contract that use  UserMintableERC721 to have this method built in.
